rand/rand.go

```diff
@@ -0,0 +1,107 @@
+package rand
+
+import (
+	"crypto/rand"
+	"fmt"
+	"math"
+	"math/big"
+)
+
+const (
+	// DefaultCharset defines the default characters
+	DefaultCharset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+
+	// DefaultLength defines the default length for random string
+	DefaultLength = 10
+)
+
+func Number() (int64, error) {
+	n, err := rand.Int(rand.Reader, big.NewInt(math.MaxInt64))
+	if err != nil {
+		return 0, fmt.Errorf("failed to generate random number: %w", err)
+	}
+	return n.Int64(), nil
+}
+
+// NumberInRange generates a random number between min and max
+func NumberInRange(min, max int64) (int64, error) {
+	if min > max {
+		return 0, fmt.Errorf("min (%d) cannot be greater than max (%d)", min, max)
+	}
+
+	rangeSize := max - min + 1
+	// Calculate the largest multiple of rangeSize that fits in MaxInt64
+	limit := math.MaxInt64 - (math.MaxInt64 % rangeSize)
+
+	for {
+		n, err := rand.Int(rand.Reader, big.NewInt(math.MaxInt64))
```

Comment:
why didn't we simply use `rand.Int(rand.Reader, big.NewInt(rangeSize))`?

Reply:
That's for generating number in a specific range that's why